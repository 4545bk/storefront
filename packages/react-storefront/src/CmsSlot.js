/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React, { Component }  from 'react'
import classnames from 'classnames'
import PropTypes from 'prop-types'
import withStyles from '@material-ui/core/styles/withStyles'

export const styles = theme => ({
  inline: {
    display: 'inline'
  },
  block: {
    display: 'block'
  }
})

/**
 * A container for HTML blob content from a CMS.  Content is dangerously inserted into the DOM.
 * Pass the html as a string as the child of this component.
 */
@withStyles(styles)
export default class CmsSlot extends Component {
  static proptypes = {
    /**
     * Use inline prop to use display:inline style
     */
    inline: PropTypes.boolean
  }
  render() {
    const { children, className, classes, inline } = this.props
    return children ? <span className={classnames(className, { [classes.inline]: inline, [classes.block]: !inline })} dangerouslySetInnerHTML={{ __html: children }} /> : null
  }
}