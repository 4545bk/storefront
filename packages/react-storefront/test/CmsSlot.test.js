/**
 * @license
 * Copyright Â© 2017-2018 Moov Corporation.  All rights reserved.
 */
import React from 'react'
import { mount } from 'enzyme'
import CmsSlot from '../src/CmsSlot'

describe('CmsSlot', () => {

  it('renders verbatim html', () => {
    const component = (
      <CmsSlot className="slot1">
        {`<p>Content Here</p>`}
      </CmsSlot>
    )

    expect(mount(component)).toMatchSnapshot()
  })

  it('renders nothing when no children are provided', () => {
    const component = (
      <CmsSlot/>
    )

    expect(mount(component)).toMatchSnapshot()
  })

})