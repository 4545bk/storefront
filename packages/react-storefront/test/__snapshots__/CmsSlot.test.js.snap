// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CmsSlot renders nothing when no children are provided 1`] = `
<WithStyles(CmsSlot)>
  <CmsSlot classes={{...}} />
</WithStyles(CmsSlot)>
`;

exports[`CmsSlot renders verbatim html 1`] = `
<WithStyles(CmsSlot) className="slot1">
  <CmsSlot className="slot1" classes={{...}}>
    <span className="slot1 CmsSlot-block-2" dangerouslySetInnerHTML={{...}} />
  </CmsSlot>
</WithStyles(CmsSlot)>
`;
